### ISSUE TITLE
XCode Command Line Tools are required (Zig bug)

### ISSUE THREAD

**Issue by slimsag** [hexops/mach#48]
Even if `(*Builder).sysroot` is set to a Mac OS SDK, the latest Zig persistently invokes `xcrun` so one must dismiss a number of prompts to install XCode and re-run the relevant Zig build command multiple times before build succeeds.

`getSDKPath` here invokes `xcrun` internally: https://sourcegraph.com/github.com/ziglang/zig@e02caa7e2909c0ae7f7e8ea1378d78e0469b9ff8/-/blob/src/main.zig?L1724-1737

It would be nice if it didn't, at least when we set `sysroot` but perhaps even without(?) so that we could supply the relevant MacOS SDKs ourselves and not have any dependency on XCode.

**Comment by slimsag** [comment]
Currently, building `mach-glfw` without XCode fails as follows:

```
slimsag@hexops-ci glfw % zig build test
Zig attempted to find the path to native system libc headers by executing this command:
cc -E -Wp,-v -xc /dev/null
Output:
==========
xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. Choose an option in the dialog to download the command line developer tools.

==========
error: unable to create compilation: CCompilerExitCode
glfw...The following command exited with error code 1:
/usr/local/bin/zig build-lib /Users/slimsag/Desktop/hexops/mach/glfw/src/main.zig -cflags -D_GLFW_COCOA -- /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/cocoa_joystick.m /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/cocoa_init.m /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/cocoa_window.m /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/cocoa_time.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/cocoa_monitor.m /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/nsgl_context.m /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/posix_thread.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/monitor.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/init.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/vulkan.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/input.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/osmesa_context.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/egl_context.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/context.c /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/src/window.c -lc -lobjc --cache-dir /Users/slimsag/Desktop/hexops/mach/glfw/zig-cache --global-cache-dir /Users/slimsag/.cache/zig --name glfw -static -isystem /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/usr/include -I /Users/slimsag/Desktop/hexops/mach/glfw/upstream/glfw/include -I /Users/slimsag/Desktop/hexops/mach/glfw/upstream/vulkan_headers/include -isystem /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/usr/include -L /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/usr/lib -L /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/usr/lib -iframework /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/System/Library/Frameworks -F /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/System/Library/Frameworks -iframework /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/System/Library/Frameworks -F /Users/slimsag/Library/Application Support/mach/sdk-macos-11.3/root/System/Library/Frameworks -framework CoreFoundation -framework Foundation -framework CoreGraphics -framework Metal -framework Cocoa -framework IOKit -framework CoreServices -framework AppKit --enable-cache 
error: the following build command failed with exit code 1:
/Users/slimsag/Desktop/hexops/mach/glfw/zig-cache/o/d260ec96d54068a71205e51c27952762/build /usr/local/bin/zig /Users/slimsag/Desktop/hexops/mach/glfw /Users/slimsag/Desktop/hexops/mach/glfw/zig-cache /Users/slimsag/.cache/zig test
```

Specifying `zig build test -Dtarget=aarch64-macos` does allow building for XCode, but one must re-run the command about 5 times and dismiss the XCode installation dialog that pops up several times.

**Comment by slimsag** [comment]
Filed https://github.com/ziglang/zig/issues/10217

**Comment by slimsag** [comment]
Fixed in Zig ziglang/zig#10217
